history-service/src/storage/attempt_store.ts

```diff
@@ -145,6 +146,58 @@ class AttemptStore implements IAttemptStore {
       totalCount,
     };
   }
+
+  async removeHistoryOwner(userId: number): Promise<void> {
+    // Delete HistoryOwner and retrieve attempts
+    const resultAttemptId: { attempt_id: number; }[] = (await this.dbConn
+      .getDataSource()
+      .createQueryBuilder()
+      .delete()
+      .from(HistoryOwnerEntity)
+      .where('user_id = :userId', { userId })
+      .returning('attempt_id')
+      .execute()
+    ).raw;
+    if (!resultAttemptId) {
+      return;
+    }
+    const deletedUserAttempts = resultAttemptId.map(
+      (res: { attempt_id: number; }) => res.attempt_id,
+    );
+    if (!deletedUserAttempts.length) {
```

Comment:
Minor nit, I think an explicit check is clearer.
```suggestion
    if (deletedUserAttempts.length === 0) {
```